docs/proposals/config_data.md

```diff
@@ -0,0 +1,319 @@
+```json
+{
+    "apiVersion": "v1",
+    "name": "PodConfigurationVol",
+    "kind": "ConfigData",
+    "data": {
+        "component_config": "COMPONENT_CONFIG_STRING_CONTENTS",
+        "component_metadata": "COMPONENT_METADATA_STRING_CONTENTS"
+    }
+}
+```
+
+Then, configuration objects are persisted to the API server as follows:
+```bash
+kubectl create -f pod-configuration-vol.json 
+```
+
+ExampleReplicationController starts 2 replicas that runs the `busybox` container.
+In this use case, the ``ConfigData`` object will be mounted as a volume and contain
+two files named "config.yaml" and "metadata.json".
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: gcr.io/google_containers/busybox
+        ports:
+        - containerPort: 80
+        volumeMounts:
+        - name: configDataVol
+          downwardAPI:
+            items:
+              - path: "config.yaml"
+                configDataRef:
+                  from: PodConfigurationVol
+                  fieldPath: component_config
+              - path: "metadata.json"
+                configDataRef:
+                  from: PodConfigurationVol
+                  fieldPath: component_metadata
+```
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ExampleReplicationController will use the ``PodConfigurationVol`` objects when starting/replicating pods.
+
+``PodConfigurationVol`` object will be watched for any modifications and the mounted volumes will be updated if there are any modifications.
+
+#### Consuming ConfigData as environment variables
+
+```json
+{
+    "apiVersion": "v1",
+    "name": "PodConfigurationEnv",
+    "kind": "ConfigData",
+    "data": {
+        "api_server": "10.240.13.14:8998",
+        "api_version": "v1",
+        "db_host": "10.240.55.3",
+        "db_port": "3306"
+    }
+}
+```
+``PodConfigurationEnv`` is intended to be consumed as environment variable key/value pairs.
+
+We create our ``ConfigData`` object.
+```bash
+kubectl create -f pod-configuration-env.json
+```
+
+ExampleReplicationController starts 2 replicas that runs the `busybox` container.
+In this use case, the ``ConfigData`` object will be consumed as 4 environment variables.
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: gcr.io/google_containers/busybox
+        ports:
+        - containerPort: 80
+        env:
+        - name: API_SERVER
+          valueFrom:
+            configDataRef:
+              from: PodConfigurationEnv
```

Comment:
I think this would be clearer if the name was pod-configuration-env or 'my-config'.  CamelCase makes it read like an API resource.